### DISCUSSION TITLE
How to use with_bandwidth_metrics in the new SwarmBuilder with a optional registry?

### DISCUSSION THREAD

**Discussion by gurinderu** [libp2p/rust-libp2p#5014] [Q&A] [answered]
I have an optional registry in our system like `registry: Option<&mut Registry>`. 
I can't understand how to use a new swarm builder with optional registry for with_bandwidth_metrics because of the types.

**Comment by thomaseizinger** [comment, marked as answer]
Your best bet is to branch on the existence of the registry and finish the builder so both branches of your `match` result in the same type. We do the same thing for our interop tests:

https://github.com/libp2p/rust-libp2p/blob/bdcb081b8860f302404c11b7c3dde393f4c38ba4/interop-tests/src/arch.rs#L38-L164

**Reply by gurinderu** [reply]
Got it, thanks
